src/fmi4c.c

```diff
@@ -1481,17 +1491,17 @@ bool loadFunctionsFmi1(fmiHandle *fmu)
 
     char dllPath[FILENAME_MAX];
     dllPath[0] = '\0';
-    strcat(dllPath, fmu->unzippedLocation);
+    strncat(dllPath, fmu->unzippedLocation, strlen(fmu->unzippedLocation));
```

Comment:
I think the use of the length parameter is incorrect for these calls to strncat. 
The purpose is to prevent overflow in the receiving (destination) buffer, in this case dllPath.
The max allowed characters to copy should be based on the remaining space in the destination buffer. Found this example on stackoverflow `strncat(dest, "values", total_size_of_dest_buffer - strlen(dest) - 1 );` (might be possible to use sizeof dllPath if its an array but I am not sure)

Right now strncat will do the same thing as strcat
This is a problem in all of the strncat calls in this PR

Reply:
I just realize that in the case of non-string-literal source, then of course the size of the data in the source buffer must also be considered, unless it is null terminated.